java-client/src/main/java/co/elastic/clients/transport/ElasticsearchTransportBase.java

```diff
@@ -243,25 +243,20 @@ private <RequestT, ResponseT, ErrorT> TransportHttpClient.Request prepareTranspo
         Map<String, String> params = endpoint.queryParameters(request);
 
         List<ByteBuffer> bodyBuffers = null;
-        HeaderMap headers = DefaultHeaders;
+        HeaderMap headers = JsonContentTypeHeaders;
```

Comment:
Please add a note here explaining why we set the content-type even for requests with no body. It will be useful for our future selves!